components/director/internal/domain/document/resolver.go

```diff
@@ -4,27 +4,67 @@ import (
 	"context"
 
 	"github.com/kyma-incubator/compass/components/director/internal/graphql"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
 )
 
-type DocumentService interface{}
+//go:generate mockery -name=DocumentService -output=automock -outpkg=automock -case=underscore
+type DocumentService interface {
+	Create(ctx context.Context, applicationID string, in model.DocumentInput) (string, error)
+	Get(ctx context.Context, id string) (*model.Document, error)
+	Delete(ctx context.Context, id string) error
+}
 
-type DocumentConverter interface{}
+//go:generate mockery -name=DocumentConverter -output=automock -outpkg=automock -case=underscore
+type DocumentConverter interface {
+	ToGraphQL(in *model.Document) *graphql.Document
+	InputFromGraphQL(in *graphql.DocumentInput) *model.DocumentInput
+}
 
 type Resolver struct {
 	svc       DocumentService
 	converter DocumentConverter
 }
 
-func NewResolver(svc DocumentService) *Resolver {
+func NewResolver(svc DocumentService, frConverter FetchRequestConverter) *Resolver {
 	return &Resolver{
 		svc:       svc,
-		converter: &converter{},
+		converter: &converter{frConverter: frConverter},
 	}
 }
 
 func (r *Resolver) AddDocument(ctx context.Context, applicationID string, in graphql.DocumentInput) (*graphql.Document, error) {
-	panic("not implemented")
+	convertedIn := r.converter.InputFromGraphQL(&in)
+
+	id, err := r.svc.Create(ctx, applicationID, *convertedIn)
+	if err != nil {
+		return nil, err
+	}
+
+	document, err := r.svc.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	gqlDocument := r.converter.ToGraphQL(document)
+
+	return gqlDocument, nil
 }
+
 func (r *Resolver) DeleteDocument(ctx context.Context, id string) (*graphql.Document, error) {
-	panic("not implemented")
+	document, err := r.svc.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if document == nil {
```

Comment:
it seems that we are not consistent here, in other resolvers we don't have special if for nil object.
Maybe service should return error if object does not exist? 
We need to have clear documentation:
- if a repository returns an error if an object does not exist?
- if a service returns an error if an object does not exist?

Please check if we are consistent among whole project

Reply:
This if statement was supposed to be removed. We discussed that case earlier with @pkosiec and we decided that repository will return error when trying to get not existing item, so the scenario where get returns nil wouldn't actually occur and that if statement wasn't needed. I checked consistency in runtime, application, document, webhook and fetchrequest domains.